src/Tests/InstrEngineTests/InstrEngineTests/TargetAppCompiler.cs

```diff
@@ -103,14 +105,14 @@ private static void AssembleILTestCode(string directoryPath)
 
                 List<IEmbeddedResourceFile> embeddedResources = ilPrefixes.Select(prefix => EmbeddedResourceUtils.GetTestResourceFile(FormattableString.Invariant($"{prefix}.il"), resourcesPath: EmbeddedResourceUtils.InvalidCSharp_EmbeddedResourcesPath)).ToList();
 
+                string assemblyName = ilPrefixes[0];
```

Comment:
>string assemblyName = ilPrefixes[0] [](http://example.com/codeflow?start=16&length=35)

Please add some comments about why we build only one, and how we choose the name of the assembly.